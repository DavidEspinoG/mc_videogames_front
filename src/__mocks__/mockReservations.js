export const mockReservation = {
  id: 26,
  user_id: 7,
  videogame_id: 68,
  days: 11,
  total_price: '220.0',
  created_at: '2023-07-20T17:53:13.512Z',
  updated_at: '2023-07-20T17:53:13.512Z',
  videogame: {
    id: 68,
    name: 'Street Fighter',
    photo: 'https://mms.businesswire.com/media/20230122005013/en/1692292/5/p1-1.jpg',
    description:
      'Street Fighter is a 1987 arcade game developed by Capcom. It is the first competitive fighting game produced by the company and the inaugural game in the Street Fighter series.',
    price_per_day: '20.0',
    created_at: '2023-07-20T17:37:47.367Z',
    updated_at: '2023-07-20T17:37:47.367Z',
  },
};

const mockReservations = [
  mockReservation,
  {
    id: 19,
    user_id: 7,
    videogame_id: 62,
    days: 2,
    total_price: '10.0',
    created_at: '2023-07-20T17:37:47.766Z',
    updated_at: '2023-07-20T17:37:47.766Z',
    videogame: {
      id: 62,
      name: 'Tetris',
      photo:
        'https://upload.wikimedia.org/wikipedia/en/thumb/7/7d/Tetris_NES_cover_art.jpg/220px-Tetris_NES_cover_art.jpg',
      description:
        'Tetris is a puzzle video game created in 1984 by Alexey Pajitnov, a Soviet software engineer.',
      price_per_day: '5.0',
      created_at: '2023-07-20T17:37:47.068Z',
      updated_at: '2023-07-20T17:37:47.068Z',
    },
  },
];

export default mockReservations;
