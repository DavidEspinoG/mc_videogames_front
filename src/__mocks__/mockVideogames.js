export const mockVideogame = {
  id: 95,
  name: 'Street Fighter',
  photo: 'https://mms.businesswire.com/media/20230122005013/en/1692292/5/p1-1.jpg',
  description:
    'Street Fighter is a 1987 arcade game developed by Capcom. It is the first competitive fighting game produced by the company and the inaugural game in the Street Fighter series.',
  price_per_day: '20.0',
  created_at: '2023-07-20T21:47:16.294Z',
  updated_at: '2023-07-20T21:47:16.294Z',
};

const mockVideogames = [
  mockVideogame,
  {
    id: 94,
    name: 'Mortal Kombat',
    photo: 'https://thesource.com/wp-content/uploads/2017/10/862771.jpg',
    description:
      'Mortal Kombat is an American media franchise centered on a series of video games, originally developed by Midway Games in 1992.',
    price_per_day: '20.0',
    created_at: '2023-07-20T21:47:16.268Z',
    updated_at: '2023-07-20T21:47:16.268Z',
  },
];

export default mockVideogames;
