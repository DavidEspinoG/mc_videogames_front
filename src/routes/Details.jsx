import { useEffect } from 'react';
import { BiGame } from 'react-icons/bi';
import { MdKeyboardArrowRight } from 'react-icons/md';
import { useDispatch, useSelector } from 'react-redux';
import { Link, useParams } from 'react-router-dom';
import colorWheel from '../assets/color_wheel.png';
import { getDetails } from '../redux/slices/videogamesSlice';
import { selectVideogameDetails } from '../redux/store';
import '../styles/details.scss';

const Details = () => {
  const dispatch = useDispatch();
  const { id } = useParams();
  const videogameDetails = useSelector(selectVideogameDetails);

  useEffect(() => {
    dispatch(getDetails(id));
  }, [dispatch, id]);

  if (!videogameDetails) {
    return <h1>Loading...</h1>;
  }

  const {
    name, photo, description, price_per_day: pricePerDay,
  } = videogameDetails;

  return (
    <section className="details-section">
      <img src={photo} alt={name} className="thumbnail" />
      <div className="details">
        <h2 className="title">{name}</h2>
        <p className="description">{description}</p>
        <table className="pricing-table">
          <tbody>
            <tr>
              <td>
                Price per day:
                {' '}
                <span>
                  $
                  {pricePerDay}
                </span>
              </td>
            </tr>
            <tr>
              <td>
                Price per week:
                {' '}
                <span>
                  $
                  {pricePerDay * 7}
                </span>
              </td>
            </tr>
            <tr>
              <td>
                Price per month:
                {' '}
                <span>
                  $
                  {pricePerDay * 30}
                </span>
              </td>
            </tr>
          </tbody>
        </table>
        <Link to="/" className="discover-button">
          Discover more videogames
          {' '}
          <span>&gt;</span>
        </Link>
        <div className="image-container">
          <BiGame className="icon" />
          <img src={colorWheel} alt="color wheel" className="color-wheel" />
        </div>
        <Link to={`/reserve/?videogameId=${id}`} className="reserve-button">
          Reserve
          <MdKeyboardArrowRight className="icon" />
        </Link>
      </div>
    </section>
  );
};

export default Details;
