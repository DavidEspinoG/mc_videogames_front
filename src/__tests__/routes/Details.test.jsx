import axios from 'axios';
import { useParams } from 'react-router-dom';
import Details from '../../routes/Details';
import { mockVideogame } from '../../__mocks__/mockVideogames';
import renderWithProviders from '../test-utils';

describe('Details', () => {
  it('should render as expected', () => {
    useParams.mockReturnValue({ id: '1' });
    const { container } = renderWithProviders(<Details />, {
      preloadedState: {
        videogames: {
          details: mockVideogame,
        },
      },
    });

    axios.get.mockResolvedValue({ status: 200, data: mockVideogame });
    expect(container).toMatchSnapshot();
  });
});
