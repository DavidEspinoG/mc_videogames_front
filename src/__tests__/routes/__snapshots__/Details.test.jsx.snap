// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Details > should render as expected 1`] = `
<div>
  <section
    class="details-section"
  >
    <img
      alt="Street Fighter"
      class="thumbnail"
      src="https://mms.businesswire.com/media/20230122005013/en/1692292/5/p1-1.jpg"
    />
    <div
      class="details"
    >
      <h2
        class="title"
      >
        Street Fighter
      </h2>
      <p
        class="description"
      >
        Street Fighter is a 1987 arcade game developed by Capcom. It is the first competitive fighting game produced by the company and the inaugural game in the Street Fighter series.
      </p>
      <table
        class="pricing-table"
      >
        <tbody>
          <tr>
            <td>
              Price per day:
               
              <span>
                $
                20.0
              </span>
            </td>
          </tr>
          <tr>
            <td>
              Price per week:
               
              <span>
                $
                140
              </span>
            </td>
          </tr>
          <tr>
            <td>
              Price per month:
               
              <span>
                $
                600
              </span>
            </td>
          </tr>
        </tbody>
      </table>
      <a
        class="discover-button"
        href="/"
      >
        Discover more videogames
         
        <span>
          &gt;
        </span>
      </a>
      <div
        class="image-container"
      >
        <svg
          class="icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.937 7.53C19.227 4.119 15.803 2 12 2 6.486 2 2 6.486 2 12s4.486 10 10 10c3.803 0 7.227-2.119 8.937-5.53a1 1 0 0 0-.397-1.316L15.017 12l5.522-3.153c.461-.264.636-.842.398-1.317zm-8.433 3.602a.999.999 0 0 0 0 1.736l6.173 3.525A7.949 7.949 0 0 1 12 20c-4.411 0-8-3.589-8-8s3.589-8 8-8a7.95 7.95 0 0 1 6.677 3.606l-6.173 3.526z"
          />
          <circle
            cx="11.5"
            cy="7.5"
            r="1.5"
          />
        </svg>
        <img
          alt="color wheel"
          class="color-wheel"
          src="/src/assets/color_wheel.png"
        />
      </div>
      <a
        class="reserve-button"
        href="/reserve?videogameId=1"
      >
        Reserve
        <svg
          class="icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
          />
        </svg>
      </a>
    </div>
  </section>
</div>
`;
